xbmc/Application.cpp

```diff
@@ -2778,6 +2779,27 @@ bool CApplication::OnMessage(CGUIMessage& message)
     }
     break;
 
+    case GUI_MSG_PLAY_TRAILER:
+    {
+      const CFileItemPtr item = std::static_pointer_cast<CFileItem>(message.GetItem());
+      std::unique_ptr<CFileItem> trailerItem =
+          ContentUtils::GeneratePlayableTrailerItem(item, g_localizeStrings.Get(20410));
```

Comment:
Is it really necessary that the first parameter of `ContentUtils::GeneratePlayableTrailerItem` is a shared pointer? Could it be a `const CFileItem&`? This would safe us the step to wrap the raw pointer contained in the message to a shared pointer.

Reply:
At this point the item (at least for current usages) is already wrapped but I understand your point since `GeneratePlayableTrailerItem` isthis is placed in utils. Adjusted as per your suggestion, hopefully done right :)